core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -168,15 +185,23 @@ trait PartitionedDataSourceScan extends PrunedUnsafeFilteredScan {
 }
 
 private[sql] final case class ZipPartitionScan(basePlan: SparkPlan with CodegenSupport,
-    otherPlan: SparkPlan, otherRDD: RDD[InternalRow] = null,
-    relation: PartitionedDataSourceScan = null) extends LeafExecNode with CodegenSupport {
+    basePartKeys: Seq[Expression],
+    otherPlan: SparkPlan,
+    otherPartKeys: Seq[Expression],
+    otherRDD: Option[RDD[InternalRow]] = None,
+    relation: Option[PartitionedDataSourceScan] = None) extends LeafExecNode with CodegenSupport {
```

Comment:
On second thought, LeafExecNode is more appropriate because it acts as a last node. BinaryExecNode is more like a binary operator which this isn't. What do you say ?